lib/isolator/orm_adapters/active_support_subscriber.rb

```diff
@@ -9,8 +9,8 @@ module ActiveSupportSubscriber
 
     def self.subscribe!(event)
       ::ActiveSupport::Notifications.subscribe(event) do |_name, _start, _finish, _id, query|
-        Isolator.incr_transactions! if START_PATTERN.match?(query[:sql])
-        Isolator.decr_transactions! if FINISH_PATTERN.match?(query[:sql])
+        Isolator.incr_transactions!(query[:connection]) if START_PATTERN.match?(query[:sql])
```

Comment:
This doesn't work for Rails <6 (see Travis failures).

Try using `query[:connection_id]` for older Rails and `query[:connection].id` or similar for newer.

Using IDs instead of connection objects is also better, 'cause it would prevent connection objects from leaking (in case of dynamically created connections).

Reply:
I have a few concerns with using connection id

1. Is it possible for connections to disconnect and reconnect with different ids? We need to reconfigure the right thresholds. It may be easier for the primary but not the non-primary connections.

2. I think it's possible for a model to establish a new connection with the same primary db. But the additional connection will be seen as a non-primary connection and get assigned the wrong threshold.

3. Are you worried about backward compatibilities for `Isolator.incr_transactions!` and `Isolator.decr_transactions!` methods? I'm using the connection and default input `connection = ActiveRecord::Base.connection` for this reason. Technically we can use `nil` id key for the default connection but that feels like code smell to me.